### PR TITLE
CVS-195132： Fix the infer crash caused by attention mask size mismatch with generate var

### PR DESCRIPTION
**Pull request by yzhou51**
### Details:

When WebNN used max_length=2048, Chromium's GPU process could terminate during the second stateful inference call. The same model worked with max_length=1024.

The failure occurred in the OpenVINO NPUW generate path. NPUW may compile multiple generate-model variants with different KV-cache capacities. For example, with:
```plaintext
NPUW_LLM_MAX_PROMPT_LEN = 2048
NPUW_LLM_MIN_RESPONSE_LEN = 128
```
the generate pyramid can select a smaller variant, such as a capacity of **1152**, for a short prompt. The caller may still provide an attention_mask sized to the application-level **max_length**, such as **2048**.

NPUW previously copied:
```c++
attention_mask->get_size() - input_tokens_len
```
elements into the selected variant's internal attention-mask tensor without checking the destination capacity. When the caller's mask was larger than the selected generate variant, this wrote beyond the destination buffer and caused memory corruption, eventually terminating Chromium's GPU process.

`max_length=1024` appeared to work because the 1024-element mask fit within the selected 1152-element generate variant.

### WebNN

- webNN demo: https://10.239.115.25:8081/demos/text-generation/?provider=webnn&model=qwen3128pipcse&verbose=1&ort=test-sym-no-zp-0806&max_length=2048&enable_causallm=1&devicetype=npu
- chromium:  
  - binary: http://powerbuilder.sh.intel.com/people/wangwei/dynamic_llm_npu/dynamic_official_no_zp_0905.packed.7z
  - commands
```shell
chrome.exe --webnn-ort-library-path-for-testing="C:\Program Files\Onnxruntime-OpenVINO" --allow-third-party-modules --webnn-ort-ep-library-path-for-testing=OpenVINOExecutionProvider?"C:\Program Files\Onnxruntime-OpenVINO\onnxruntime_providers_openvino_plugin.dll" --enable-features=WinSboxHighGPUJobMemoryLimits,WebMachineLearningNeuralNetwork,WebNNOnnxRuntime --no-sandbox --webnn-ort-logging-level=VERBOSE --webnn-ort-ovep-enable-causallm --webnn-ort-disable-cpu-fallback
```
- openvino: 
  - Repo: https://github.com/openvinotoolkit/openvino 
  - code base: bb19db76fa5f1661dc7976a9c93def1ce09a66f7
  - build: Release
- onnxruntim: 
  - Repo: https://github.com/intel-innersource/frameworks.ai.onnxruntime.staging 
  - code base : d46033d92626b95e1daddda09052f0f24374ca9e
  - build: Release
- OpenVINO EP 
  - Repo: https://github.com/intel-innersource/frameworks.ai.onnxruntime.openvino-plugin-ep
  - code base: ddf2caee5cc64083afd0390010db3307ffe72245
  - build: Release
### Tickets:
 - [CVS-195132](https://jira.devtools.intel.com/browse/CVS-195132)

### AI Assistance:
 - *AI assistance used: yes*
 - *If yes, summarize how AI was used and what human validation was performed (build/tests/manual checks).*
   - AI help generate the fixing.
   - human: debug and nalysis the issue, root cause the issue.

### CODE REVIEW

**Comment by yzhou51** [comment]
@cwzrad @dylanneve1 Could you help review this? Thanks!

**Review by intelgaoxiong — COMMENTED** [review]
This clamp fixes the OOB write, but I’d like to confirm the invariant that makes it correct.

`select_generate_request()` chooses the smallest generate variant whose capacity can fit the effective prompt+response budget. However, here we copy:
`attention_mask->get_size() - input_tokens_len`
elements into the selected variant’s internal KV-mask. That value is the caller-side buffer length, not necessarily the number of valid historical tokens. If the caller allocates a larger fixed-size mask (for example `max_length=2048`) while the selected generate variant only has capacity for a smaller past window, this code silently truncates the mask.

Before we treat this as valid behavior, please confirm that:
1. the valid past-token portion of `attention_mask` always fits in the selected variant, and
2. any truncated tail is guaranteed to be padding / not-valid-context and not real attention history.

If the valid context can exceed the selected variant capacity, then the right fix is to select/switch to a larger variant (or fail explicitly), not to silently clamp the copy.
